### PR TITLE
refactor machine

### PR DESCRIPTION
**Pull request by wlan0**
@cjellick @will-chan

### CODE REVIEW

**Comment by wlan0** [comment]
This fixes 
https://github.com/rancherio/rancher/issues/809

**Comment by cjellick** [comment]
@wlan0 Did you run (at least some) of the driver-specific tests locally for this pr?

**Comment by cjellick** [comment]
And also did you run validation tests against it?

**Comment by cjellick** [comment]
For the record, this pr _does not_ address rancherio/rancher#1048

**Comment by wlan0** [comment]
@cjellick Ran both local tests and validation tests. Both pass.

File: handlers/common.go [hunk 1]
```diff
@@ -116,3 +141,121 @@ func initEnviron(machineDir string) []string {
...
+		_, err = io.Copy(tarfileWriter, file)
+		if err != nil {
+			return "", err
+		}
+	}
+	return destFile, nil
+}
+
+func getIdempotentExtractedConfig(machine *client.Machine, machineDir string, apiClient *client.RancherClient) (string, error) {
+	if machine.Driver == "virtualbox" {
```
Comment (will-chan) [inline handlers/common.go]: Can we extract this to a common function as the previous check in createExtractedConfig.

**Comment by will-chan** [comment]
LGTM other than the one common function fix.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**